src/icon-button.tsx

```diff
@@ -1,9 +1,63 @@
+'use client';
 import * as React from 'react';
-import classnames from 'classnames';
 import {BaseTouchable} from './touchable';
 import * as styles from './icon-button.css';
+import classNames from 'classnames';
+import Spinner from './spinner';
+import {useThemeVariant} from './theme-variant-context';
 
-import type {DataAttributes, TrackingEvent} from './utils/types';
+import type {TouchableElement} from './touchable';
+import type {ExclusifyUnion} from './utils/utility-types';
+import type {DataAttributes, IconProps, TrackingEvent} from './utils/types';
+
+interface HrefProps {
+    href: string;
+    newTab?: boolean;
+}
+
+interface ToProps {
+    to: string;
+    fullPageOnWebView?: boolean;
+    replace?: boolean;
+}
+
+interface OnPressProps {
+    onPress: (event: React.MouseEvent<HTMLElement>) => void | undefined | Promise<void>;
+}
+
+interface MaybeProps {
+    onPress?: undefined;
+    href?: undefined;
+    to?: undefined;
+}
+
+interface BaseOldProps {
+    /** @deprecated */
+    children?: React.ReactNode;
+    /** @deprecated */
+    className?: string;
+    disabled?: boolean;
+    /** @deprecated */
+    icon?: string;
+    /** @deprecated */
+    iconSize?: number;
+    /** @deprecated */
+    backgroundColor?: string;
+    /** @deprecated */
+    size?: number | string;
+    /** @deprecated */
+    style?: React.CSSProperties;
+    /** @deprecated */
+    'aria-live'?: 'polite' | 'off' | 'assertive';
+
+    trackingEvent?: TrackingEvent | ReadonlyArray<TrackingEvent>;
+    /** "data-" prefix is automatically added. For example, use "testid" instead of "data-testid" */
+    dataAttributes?: DataAttributes;
+    newTab?: boolean;
+    'aria-label'?: string;
+}
+
+type OldProps = BaseOldProps & ExclusifyUnion<HrefProps | ToProps | OnPressProps | MaybeProps>;
 
 const ICON_SIZE_1 = 24;
```

Comment:
what is "1" in this name? 🤔

Reply:
This is old code. I left everything related to old implementation untouched